// @ng-bootstrap/ng-bootstrap/accordion/accordion.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@angular/common';
import * as import2 from '@ng-bootstrap/ng-bootstrap/accordion/accordion';
import * as import3 from '@ng-bootstrap/ng-bootstrap/accordion/accordion-config';
const styles_NgbAccordion:any[] = ([] as any[]);
export const RenderType_NgbAccordion:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_NgbAccordion,
  data: {}
}
);
function View_NgbAccordion_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[(l()(),import0.ɵand(0,(null as any),(null as any),0))],(null as any),(null as any));
}
function View_NgbAccordion_4(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[(l()(),import0.ɵand(0,(null as any),(null as any),0))],(null as any),(null as any));
}
function View_NgbAccordion_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),4,'div',[
      [
        'class',
        'card-block'
      ]
      ,
      [
        'role',
        'tabpanel'
      ]

    ]
    ,[
      [
        8,
        'id',
        0
      ]
      ,
      [
        1,
        'aria-labelledby',
        0
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_NgbAccordion_4)),
      import0.ɵdid(270336,(null as any),0,import1.NgTemplateOutlet,[import0.ViewContainerRef],{ngTemplateOutlet: [
        0,
        'ngTemplateOutlet'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n      ']))
  ]
  ,(ck,v) => {
    const currVal_2:any = v.parent.context.$implicit.contentTpl.templateRef;
    ck(v,3,0,currVal_2);
  },(ck,v) => {
    const currVal_0:any = import0.ɵinlineInterpolate(1,'',v.parent.context.$implicit.id,'');
    const currVal_1:any = (v.parent.context.$implicit.id + '-header');
    ck(v,0,0,currVal_0,currVal_1);
  });
}
function View_NgbAccordion_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n      '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),7,'div',[[
        'role',
        'tab'
      ]
    ],[
      [
        8,
        'id',
        0
      ]
      ,
      [
        1,
        'aria-selected',
        0
      ]
      ,
      [
        8,
        'className',
        0
      ]
      ,
      [
        2,
        'active',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),4,'a',[[
        'href',
        ''
      ]
    ],[
      [
        2,
        'text-muted',
        (null as any)
      ]
      ,
      [
        1,
        'aria-expanded',
        0
      ]
      ,
      [
        1,
        'aria-controls',
        0
      ]

    ]
    ,[
      [
        (null as any),
        'click'
      ]
      ,
      [
        (null as any),
        'focus'
      ]
      ,
      [
        (null as any),
        'blur'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>!!co.toggle(v.context.$implicit.id)) !== false);
        ad = (pd_0 && ad);
      }
      if (('focus' === en)) {
        const pd_1:any = ((<any>(v.context.$implicit.focused = true)) !== false);
        ad = (pd_1 && ad);
      }
      if (('blur' === en)) {
        const pd_2:any = ((<any>(v.context.$implicit.focused = false)) !== false);
        ad = (pd_2 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '\n          ',
      ''
    ]
    )),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_NgbAccordion_2)),
      import0.ɵdid(270336,(null as any),0,import1.NgTemplateOutlet,[import0.ViewContainerRef],{ngTemplateOutlet: [
        0,
        'ngTemplateOutlet'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n      '])),
    (l()(),import0.ɵted((null as any),['\n      '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_NgbAccordion_3)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_8:any = ((v.context.$implicit.titleTpl == null)? (null as any): v.context.$implicit.titleTpl.templateRef);
    ck(v,6,0,currVal_8);
    const currVal_9:any = co.isOpen(v.context.$implicit.id);
    ck(v,11,0,currVal_9);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = import0.ɵinlineInterpolate(1,'',v.context.$implicit.id,'-header');
    const currVal_1:any = v.context.$implicit.focused;
    const currVal_2:any = ('card-header ' + (v.context.$implicit.type? ('card-' + v.context.$implicit.type): (co.type? ('card-' + co.type): '')));
    const currVal_3:any = co.isOpen(v.context.$implicit.id);
    ck(v,1,0,currVal_0,currVal_1,currVal_2,currVal_3);
    const currVal_4:any = v.context.$implicit.disabled;
    const currVal_5:any = co.isOpen(v.context.$implicit.id);
    const currVal_6:any = v.context.$implicit.id;
    ck(v,3,0,currVal_4,currVal_5,currVal_6);
    const currVal_7:any = v.context.$implicit.title;
    ck(v,4,0,currVal_7);
  });
}
export function View_NgbAccordion_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n  '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),4,'div',[[
        'class',
        'card'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_NgbAccordion_1)),
    import0.ɵdid(401408,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n  '])),
    (l()(),import0.ɵted((null as any),['\n']))
  ]
  ,(ck,v) => {
    var co:import2.NgbAccordion = v.component;
    const currVal_0:any = co.panels;
    ck(v,4,0,currVal_0);
  },(null as any));
}
function View_NgbAccordion_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'ngb-accordion',[[
        'role',
        'tablist'
      ]
      ],[[
        1,
        'aria-multiselectable',
        0
      ]
    ],(null as any),(null as any),View_NgbAccordion_0,RenderType_NgbAccordion)),
    import0.ɵdid(1073152,(null as any),1,import2.NgbAccordion,[import3.NgbAccordionConfig],(null as any),(null as any)),
    import0.ɵqud(301989888,1,{panels: 1})
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:boolean = !import0.ɵnov(v,1).closeOtherPanels;
    ck(v,0,0,currVal_0);
  });
}
export const NgbAccordionNgFactory:import0.ComponentFactory<import2.NgbAccordion> = import0.ɵccf('ngb-accordion',import2.NgbAccordion,View_NgbAccordion_Host_0,{
  activeIds: 'activeIds',
  closeOtherPanels: 'closeOthers',
  type: 'type'
}
,{panelChange: 'panelChange'},([] as any[]));
