// @ng-bootstrap/ng-bootstrap/progressbar/progressbar.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@angular/common';
import * as import2 from '@ng-bootstrap/ng-bootstrap/progressbar/progressbar';
import * as import3 from '@ng-bootstrap/ng-bootstrap/progressbar/progressbar-config';
const styles_NgbProgressbar:any[] = ([] as any[]);
export const RenderType_NgbProgressbar:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_NgbProgressbar,
  data: {}
}
);
function View_NgbProgressbar_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      '%'
    ]
    ))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.getPercentValue();
    ck(v,1,0,currVal_0);
  });
}
export function View_NgbProgressbar_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(2,[
    (l()(),import0.ɵted((null as any),['\n    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),8,'div',[[
        'class',
        'progress'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n      '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),5,'div',[
      [
        'aria-valuemin',
        '0'
      ]
      ,
      [
        'role',
        'progressbar'
      ]

    ]
    ,[
      [
        8,
        'className',
        0
      ]
      ,
      [
        4,
        'width',
        '%'
      ]
      ,
      [
        1,
        'aria-valuenow',
        0
      ]
      ,
      [
        1,
        'aria-valuemax',
        0
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_NgbProgressbar_1)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    import0.ɵncd((null as any),0),
    (l()(),import0.ɵted((null as any),['\n      '])),
    (l()(),import0.ɵted((null as any),['\n    '])),
    (l()(),import0.ɵted((null as any),['\n  ']))
  ]
  ,(ck,v) => {
    var co:import2.NgbProgressbar = v.component;
    const currVal_4:any = co.showValue;
    ck(v,6,0,currVal_4);
  },(ck,v) => {
    var co:import2.NgbProgressbar = v.component;
    const currVal_0:any = import0.ɵinlineInterpolate(3,'progress-bar',(co.type? (' bg-' + co.type): ''),'',(co.animated? ' progress-bar-animated': ''),'',(co.striped? ' progress-bar-striped': ''),'');
    const currVal_1:any = co.getPercentValue();
    const currVal_2:any = co.getValue();
    const currVal_3:any = co.max;
    ck(v,3,0,currVal_0,currVal_1,currVal_2,currVal_3);
  });
}
function View_NgbProgressbar_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'ngb-progressbar',([] as any[]),(null as any),(null as any),(null as any),View_NgbProgressbar_0,RenderType_NgbProgressbar)),
    import0.ɵdid(24576,(null as any),0,import2.NgbProgressbar,[import3.NgbProgressbarConfig],(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const NgbProgressbarNgFactory:import0.ComponentFactory<import2.NgbProgressbar> = import0.ɵccf('ngb-progressbar',import2.NgbProgressbar,View_NgbProgressbar_Host_0,{
  max: 'max',
  animated: 'animated',
  striped: 'striped',
  showValue: 'showValue',
  type: 'type',
  value: 'value'
}
,{},['*']);
