// @ng-bootstrap/ng-bootstrap/pagination/pagination.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@angular/common';
import * as import2 from '@ng-bootstrap/ng-bootstrap/pagination/pagination';
import * as import3 from '@ng-bootstrap/ng-bootstrap/pagination/pagination-config';
const styles_NgbPagination:any[] = ([] as any[]);
export const RenderType_NgbPagination:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_NgbPagination,
  data: {}
}
);
function View_NgbPagination_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),10,'li',[[
        'class',
        'page-item'
      ]
      ],[[
        2,
        'disabled',
        (null as any)
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n          '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),7,'a',[
      [
        'aria-label',
        'First'
      ]
      ,
      [
        'class',
        'page-link'
      ]
      ,
      [
        'href',
        ''
      ]

    ]
      ,[[
        1,
        'tabindex',
        0
      ]
      ],[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>!!co.selectPage(1)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'aria-hidden',
        'true'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['««'])),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'sr-only'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['First'])),
    (l()(),import0.ɵted((null as any),['\n          '])),
    (l()(),import0.ɵted((null as any),['                \n        ']))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:boolean = (!co.hasPrevious() || co.disabled);
    ck(v,0,0,currVal_0);
    const currVal_1:any = (co.hasPrevious()? (null as any): '-1');
    ck(v,2,0,currVal_1);
  });
}
function View_NgbPagination_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),10,'li',[[
        'class',
        'page-item'
      ]
      ],[[
        2,
        'disabled',
        (null as any)
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n          '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),7,'a',[
      [
        'aria-label',
        'Previous'
      ]
      ,
      [
        'class',
        'page-link'
      ]
      ,
      [
        'href',
        ''
      ]

    ]
      ,[[
        1,
        'tabindex',
        0
      ]
      ],[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>!!co.selectPage((co.page - 1))) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'aria-hidden',
        'true'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['«'])),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'sr-only'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['Previous'])),
    (l()(),import0.ɵted((null as any),['\n          '])),
    (l()(),import0.ɵted((null as any),['\n        ']))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:boolean = (!co.hasPrevious() || co.disabled);
    ck(v,0,0,currVal_0);
    const currVal_1:any = (co.hasPrevious()? (null as any): '-1');
    ck(v,2,0,currVal_1);
  });
}
function View_NgbPagination_4(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'a',[[
        'class',
        'page-link'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['...']))
  ]
  ,(null as any),(null as any));
}
function View_NgbPagination_5(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'a',[
      [
        'class',
        'page-link'
      ]
      ,
      [
        'href',
        ''
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>!!co.selectPage(v.parent.context.$implicit)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    ))
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = v.parent.context.$implicit;
    ck(v,1,0,currVal_0);
  });
}
function View_NgbPagination_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),7,'li',[[
        'class',
        'page-item'
      ]
    ],[
      [
        2,
        'active',
        (null as any)
      ]
      ,
      [
        2,
        'disabled',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n          '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_NgbPagination_4)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n          '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_NgbPagination_5)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_2:any = co.isEllipsis(v.context.$implicit);
    ck(v,3,0,currVal_2);
    const currVal_3:boolean = !co.isEllipsis(v.context.$implicit);
    ck(v,6,0,currVal_3);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = (v.context.$implicit === co.page);
    const currVal_1:any = (co.isEllipsis(v.context.$implicit) || co.disabled);
    ck(v,0,0,currVal_0,currVal_1);
  });
}
function View_NgbPagination_6(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),10,'li',[[
        'class',
        'page-item'
      ]
      ],[[
        2,
        'disabled',
        (null as any)
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n          '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),7,'a',[
      [
        'aria-label',
        'Next'
      ]
      ,
      [
        'class',
        'page-link'
      ]
      ,
      [
        'href',
        ''
      ]

    ]
      ,[[
        1,
        'tabindex',
        0
      ]
      ],[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>!!co.selectPage((co.page + 1))) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'aria-hidden',
        'true'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['»'])),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'sr-only'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['Next'])),
    (l()(),import0.ɵted((null as any),['\n          '])),
    (l()(),import0.ɵted((null as any),['\n        ']))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:boolean = (!co.hasNext() || co.disabled);
    ck(v,0,0,currVal_0);
    const currVal_1:any = (co.hasNext()? (null as any): '-1');
    ck(v,2,0,currVal_1);
  });
}
function View_NgbPagination_7(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),10,'li',[[
        'class',
        'page-item'
      ]
      ],[[
        2,
        'disabled',
        (null as any)
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n          '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),7,'a',[
      [
        'aria-label',
        'Last'
      ]
      ,
      [
        'class',
        'page-link'
      ]
      ,
      [
        'href',
        ''
      ]

    ]
      ,[[
        1,
        'tabindex',
        0
      ]
      ],[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>!!co.selectPage(co.pageCount)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'aria-hidden',
        'true'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['»»'])),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'sr-only'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['Last'])),
    (l()(),import0.ɵted((null as any),['\n          '])),
    (l()(),import0.ɵted((null as any),['                \n        ']))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:boolean = (!co.hasNext() || co.disabled);
    ck(v,0,0,currVal_0);
    const currVal_1:any = (co.hasNext()? (null as any): '-1');
    ck(v,2,0,currVal_1);
  });
}
export function View_NgbPagination_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(2,[
    (l()(),import0.ɵted((null as any),['\n    '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),19,'nav',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n      '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),16,'ul',([] as any[]),[[
        8,
        'className',
        0
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_NgbPagination_1)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n      \n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_NgbPagination_2)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_NgbPagination_3)),
    import0.ɵdid(401408,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_NgbPagination_6)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        \n        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_NgbPagination_7)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['        \n      '])),
    (l()(),import0.ɵted((null as any),['\n    '])),
    (l()(),import0.ɵted((null as any),['\n  ']))
  ]
  ,(ck,v) => {
    var co:import2.NgbPagination = v.component;
    const currVal_1:any = co.boundaryLinks;
    ck(v,6,0,currVal_1);
    const currVal_2:any = co.directionLinks;
    ck(v,9,0,currVal_2);
    const currVal_3:any = co.pages;
    ck(v,12,0,currVal_3);
    const currVal_4:any = co.directionLinks;
    ck(v,15,0,currVal_4);
    const currVal_5:any = co.boundaryLinks;
    ck(v,18,0,currVal_5);
  },(ck,v) => {
    var co:import2.NgbPagination = v.component;
    const currVal_0:any = ('pagination' + (co.size? (' pagination-' + co.size): ''));
    ck(v,3,0,currVal_0);
  });
}
function View_NgbPagination_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'ngb-pagination',([] as any[]),(null as any),(null as any),(null as any),View_NgbPagination_0,RenderType_NgbPagination)),
    import0.ɵdid(286720,(null as any),0,import2.NgbPagination,[import3.NgbPaginationConfig],(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const NgbPaginationNgFactory:import0.ComponentFactory<import2.NgbPagination> = import0.ɵccf('ngb-pagination',import2.NgbPagination,View_NgbPagination_Host_0,{
  disabled: 'disabled',
  boundaryLinks: 'boundaryLinks',
  directionLinks: 'directionLinks',
  ellipses: 'ellipses',
  rotate: 'rotate',
  collectionSize: 'collectionSize',
  maxSize: 'maxSize',
  page: 'page',
  pageSize: 'pageSize',
  size: 'size'
}
,{pageChange: 'pageChange'},([] as any[]));
